core/services/network_config.rb

```diff
@@ -18,7 +18,7 @@ def initialize(config, logger)
   # @param node_names [Array<String>] of node to add
   def add_nodes(node_names)
     node_names.each do |name|
-      @nodes[name] = Node.new(@config, name) if @config.node_names.include?(name)
+      @nodes[name] = Node.new(@config, name) if @config.all_node_names.include?(name)
```

Comment:
What is the point of checking for the name to be in the `all_node_name` list? The only scenario this checks seems to be feasible is when the developer will pass the wrong parameter. I do not see how it can happen.